rust/src/client.rs

```diff
@@ -0,0 +1,166 @@
+use crate::service::Handle;
+use crate::coordinator::RoundParameters;
+use crate::participant::{Participant, Task};
+use crate::{CoordinatorPublicKey, SeedDict, SumDict, PetError, InitError};
+use sodiumoxide::crypto::box_;
+use std::sync::Arc;
+use std::collections::HashMap;
+use std::future::Future;
+use std::pin::Pin;
+use tokio::time;
+use std::time::Duration;
+
+
+/// Client-side errors
+pub enum ClientError {
```

Comment:
For errors, it's convenient to use `thiserror` to derive the `::std::error::Error` trait automatically.